DAT-16091 DevOps :: Configure liquibase-hanadb repo to use build-logic

jandroav opened the pull request:
https://datical.atlassian.net/browse/DAT-16091

Commit:
#DAT-16091

Commit:
chore(pom.xml): add maven-failsafe-plugin to redirect test output to file and configure report format
feat(pom.xml): add system property variable for spock-reports output directory
feat(pom.xml): add dependency for surefire-junit47 to force using the latest JUnit 47 provider

Commit:
chore(.snyk): remove .snyk file as it is no longer needed for vulnerability management

Commit:
chore(pom.xml): update liquibase-hanadb version to 0-SNAPSHOT for development purposes
chore(pom.xml): update project name to "Liquibase Extension: SAP HANA support" for clarity
chore(pom.xml): update project description to reflect the purpose of the extension
chore(pom.xml): update organization name to "Liquibase" for consistency
chore(pom.xml): update license information to use Apache License, Version 2.0
chore(pom.xml): update issue management URL to point to the GitHub repository issues
chore(pom.xml): update SCM information to use Git and point to the GitHub repository
chore(pom.xml): update liquibase.version property to 0-SNAPSHOT for development purposes
chore(pom.xml): update project.build.sourceEncoding and project.reporting.outputEncoding to UTF-8
chore(pom.xml): update surefire-plugin version to 3.1.2 for compatibility
chore(pom.xml): update build-helper-maven-plugin version to 3.4.0 for compatibility
chore(pom.xml): update maven-javadoc-plugin version to 3.5.0 for compatibility
chore(pom.xml): update maven-source-plugin version to 3.3.0 for compatibility
chore(pom.xml): update jacoco-maven-plugin version to 0.8.11 for compatibility
chore(pom.xml): update sonar-maven-plugin version to 3.9.1.2184 for compatibility
chore(pom.xml): update liquibase-sdk-maven-plugin version to 0.10.20 for compatibility

Commit:
chore(pom.xml): update liquibase-hanadb version to 4.24.1-SNAPSHOT
chore(pom.xml): update liquibase.version property to 4.24.0
chore(pom.xml): remove unused dependencies and plugins
chore(pom.xml): update maven-surefire-plugin version to ${maven-surefire-plugin.version}
chore(pom.xml): update maven-failsafe-plugin version to ${maven-failsafe-plugin.version}

jnewton03 commented on .github/workflows/release-published.yml:
```diff
@@ -1,80 +1,52 @@
+          repository: "liquibase/liquibase-mongodb"
 
-      - name: Publish to Maven Central
+      - name: Upload to s3
```
@jandroav same thing here. I think its just copy/pasta from another extension.  We shouldn't need to upload XSDs

jandroav replied:
fixed

Commit:
chore(test.yml): add build and package step to the workflow to ensure dependencies are downloaded and project is built before running tests
chore(test.yml): create HXE directory before starting Hana to ensure it exists

Commit:
chore(workflows): update liquibase/build-logic workflows to v0.5.3

The liquibase/build-logic workflows have been updated to version v0.5.3. This update includes the following changes:

- attach-artifact-release.yml: Updated to v0.5.3
- build-nightly.yml: Updated to v0.5.3
- codeql.yml: Updated to v0.5.3
- create-release.yml: Updated to v0.5.3
- release-published.yml: Updated to v0.5.3
- test.yml: Updated to v0.5.3

These updates are necessary to ensure that the workflows are using the latest version of the build logic and take advantage of any bug fixes or improvements that have been made.

Commit:
chore(test.yml): remove unnecessary empty line to improve code readability

Commit:
chore(test.yml): add maven settings to look for dependencies in the Liquibase repositories

feat(test.yml): add Liquibase repositories and servers configuration to enable access to Liquibase dependencies

The changes in this commit add the necessary configuration to the test workflow to look for dependencies in the Liquibase repositories. This is achieved by adding a maven-settings-xml-action step that configures the repositories and servers for Liquibase. The repositories are configured with the URLs for the Liquibase and Liquibase Pro repositories, enabling access to their releases and snapshots. The servers are configured with the necessary credentials to authenticate with the Liquibase and Liquibase Pro repositories. This allows the workflow to install the liquibase-core 0-SNAPSHOT dependency.

Commit:
chore(pom.xml): add liquibase-sdk-maven-plugin to manage database schema changes

The liquibase-sdk-maven-plugin is added to the project's pom.xml file. This plugin will be used to manage database schema changes using Liquibase.

Commit:
chore(pom.xml): add configuration for liquibase-sdk-maven-plugin to include githubToken environment variable

The configuration for the liquibase-sdk-maven-plugin has been updated to include the githubToken environment variable. This allows the plugin to access the token for authentication when interacting with GitHub.

Commit:
chore(test.yml): remove unused build cache step to simplify the workflow

Commit:
chore(.DS_Store): add .DS_Store file to the repository
fix(pom.xml): remove unused liquibase-sdk-maven-plugin from pom.xml to reduce unnecessary dependencies

Commit:
chore: remove .DS_Store file

The .DS_Store file was deleted as it is an unnecessary file that is automatically generated by macOS. It is not needed for the project and can be safely removed.

Commit:
chore(pom.xml): add liquibase-sdk-maven-plugin to enable Liquibase integration with GitHub Actions
feat(pom.xml): configure the plugin to use the GITHUB_TOKEN environment variable for authentication with GitHub API

Commit:
chore(test.yml): comment out integration-test job to temporarily disable it

Commit:
chore(test.yml): remove unnecessary whitespace and commented out code
chore(test.yml): comment out the upload-artifact step for test-reports-integration to prevent unnecessary artifact uploads

Commit:
chore: update liquibase build logic workflows to version v0.5.5

chore(pom.xml): add SCM information for liquibase-hanadb repository

jandroav committed:
Merge branch 'main' into DAT-16091

jandroav committed:
Merge branch 'main' into DAT-16091
